### PR TITLE
feat: add icp as standard

### PR DESCRIPTION
**Pull request by aleacevedo**
# feat: add icp as standard

## Summary

Implement ICP wrapped to use token as any other standard

## Shortcut
https://app.shortcut.com/terminalsystems/story/24865/refactor-make-icp-standard-token-and-icp-standard-wrapper

### CODE REVIEW

**Review by angarita-dev — APPROVED** [review]

File: src/token_standards/dip20Methods.ts [hunk 1]
```diff
@@ -65,9 +65,12 @@ const burnXTC = async (
   throw new Error('BURN NOT SUPPORTED');
 };
 
+const getDecimals = async (actor: ActorSubclass<BaseDip20Service>) => getDecimalsFromMetadata(await getMetadata(actor))
```
Comment (tomiir) [inline src/token_standards/dip20Methods.ts:68]: whhy not use this function in all this file?

File: src/token_standards/extMethods.ts [hunk 2]
```diff
@@ -85,9 +85,13 @@ const burnXTC = async (
   throw new Error('BURN NOT SUPPORTED');
 };
 
+const getDecimals = async (actor: ActorSubclass<BaseExtService>) => getDecimalsFromMetadata(await getMetadata(actor))
```
Comment (tomiir) [inline src/token_standards/extMethods.ts:88]: same

File: src/token_standards/wicpMethods.ts [hunk 3]
```diff
@@ -65,9 +65,12 @@ const burnXTC = async (
   throw new Error('BURN NOT SUPPORTED');
 };
 
+const getDecimals = async (actor: ActorSubclass<BaseWICPService>) => getDecimalsFromMetadata(await getMetadata(actor))
```
Comment (tomiir) [inline src/token_standards/wicpMethods.ts:68]: same

File: src/token_standards/xtcMethods.ts [hunk 4]
```diff
@@ -61,14 +61,17 @@ const burnXTC = async (
   actor: ActorSubclass<BaseXtcService>,
   { to, amount }: BurnParams
 ): Promise<BurnResult> => {
-  const decimals = getDecimals(await getMetadata(actor));
+  const decimals = getDecimalsFromMetadata(await getMetadata(actor));
   const parsedAmount = parseAmountToSend(amount, decimals);
   return actor._burn({ canister_id: to, amount: parsedAmount });
 };
 
+const getDecimals = async (actor: ActorSubclass<BaseXtcService>) => getDecimalsFromMetadata(await getMetadata(actor))
```
Comment (tomiir) [inline src/token_standards/xtcMethods.ts:69]: same and it seems it can be abstracted

Comment (aleacevedo) [reply src/token_standards/xtcMethods.ts:69]: It cannot because it use getMetadata, We already had problems with that

**Review by tomiir — APPROVED** [review]

**Pull request merged**